### PR TITLE
Update ERC780 contract

### LINKED ISSUE #133: (Refactor) Apply updates of ERC780
- **What is it?** (leave one option)
    * `(Refactor)` proposal (i.e. no new features or updates, just code optimization/library updates)

https://github.com/ethereum/EIPs/issues/780

Unfortunately, erc780 was updated recently and possibility to delete claim about self was removed, now only the issuer can remove a claim it made previously
```
    function removeClaim(address issuer, address subject, bytes32 key) public {
        require(msg.sender == issuer);
```

We should
- [x] update erc780 code https://github.com/poanetwork/poa-popa/blob/master/blockchain/contracts/EthereumClaimsRegistry.sol
- [x] update tests https://github.com/poanetwork/poa-popa/blob/master/blockchain/test/ethereum_claims_registry_specs.js#L43
- [x] now we can decrement `totalUsers`, `totalAddresses` and `totalConfirmed` when address is unregistered since we can be sure that no one else can unregister it 
- [x] redeploy new version on sokol
- [x] redeploy new version on core

### PR DESCRIPTION
**Pull request by fvictorio**
Closes #133.

This updates the claims registry contract. I also deployed it in Rinkeby and Sokol, and updated the address of the registry in the migration file. I will deploy it in Core soon.

This is the data to deploy the contract in other chains:

```
senderAddress: 0x0ae8063ec2c2de5c2f9a6e21fa68404a201666a0
rawTx: 0xf90747808502540be4008307e3058080b906f46060604052341561000f57600080fd5b6106d68061001e6000396000f30060606040526004361061006d576000357c0100000000000000000000000000000000000000000000000000000000900463ffffffff1680639155b01a146100725780639918925d146100a6578063c7508ec7146100f9578063e1661eff1461015e578063e6d970aa146101df575b600080fd5b341561007d57600080fd5b6100a460048080356000191690602001909190803560001916906020019091905050610260565b005b34156100b157600080fd5b6100f7600480803573ffffffffffffffffffffffffffffffffffffffff16906020019091908035600019169060200190919080356000191690602001909190505061037b565b005b341561010457600080fd5b61015c600480803573ffffffffffffffffffffffffffffffffffffffff1690602001909190803573ffffffffffffffffffffffffffffffffffffffff1690602001909190803560001916906020019091905050610497565b005b341561016957600080fd5b6101c1600480803573ffffffffffffffffffffffffffffffffffffffff1690602001909190803573ffffffffffffffffffffffffffffffffffffffff16906020019091908035600019169060200190919050506105d8565b60405180826000191660001916815260200191505060405180910390f35b34156101ea57600080fd5b610242600480803573ffffffffffffffffffffffffffffffffffffffff1690602001909190803573ffffffffffffffffffffffffffffffffffffffff1690602001909190803560001916906020019091905050610678565b60405180826000191660001916815260200191505060405180910390f35b806000803373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff16815260200190815260200160002060003373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff16815260200190815260200160002060008460001916600019168152602001908152602001600020816000191690555081600019163373ffffffffffffffffffffffffffffffffffffffff163373ffffffffffffffffffffffffffffffffffffffff167f8c9f893548e8429f9352aba6698e6f4dca2e390604f6e8c5881a7a505d94ae5084426040518083600019166000191681526020018281526020019250505060405180910390a45050565b806000803373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff16815260200190815260200160002060008573ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff16815260200190815260200160002060008460001916600019168152602001908152602001600020816000191690555081600019168373ffffffffffffffffffffffffffffffffffffffff163373ffffffffffffffffffffffffffffffffffffffff167f8c9f893548e8429f9352aba6698e6f4dca2e390604f6e8c5881a7a505d94ae5084426040518083600019166000191681526020018281526020019250505060405180910390a4505050565b8273ffffffffffffffffffffffffffffffffffffffff163373ffffffffffffffffffffffffffffffffffffffff161415156104d157600080fd5b6000808473ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff16815260200190815260200160002060008373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff168152602001908152602001600020600082600019166000191681526020019081526020016000206000905580600019168273ffffffffffffffffffffffffffffffffffffffff163373ffffffffffffffffffffffffffffffffffffffff167f7bc046ef9f2fccd4c99778d7e0fbed67ddb16e63e1a266894769de9516b8039a426040518082815260200191505060405180910390a4505050565b60008060008573ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff16815260200190815260200160002060008473ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff168152602001908152602001600020600083600019166000191681526020019081526020016000205490509392505050565b6000602052826000526040600020602052816000526040600020602052806000526040600020600092509250505054815600a165627a7a72305820be9b6247293e195d4d0072a2530b6ab5805aa117b68de57d8ab18eb5121c1f1d00291ba079be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798a00bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
cost (ether): 0.00516869
contractAddress: 0xec9cd1a18ca13e8703bdbcc68419e0d08aeb3528
```

### CODE REVIEW

**Comment by fvictorio** [comment]
Deployed on Core.

File: blockchain/truffle.js [hunk 1]
```diff
@@ -27,5 +28,9 @@ module.exports = {
             provider: new HDWalletProvider(mnemonic, rinkebyUrl),
             network_id: 4,
         },
+        sokol: {
+            provider: new HDWalletProvider(mnemonic, sokolUrl),
+            network_id: 4,
```
Comment (phahulin) [inline blockchain/truffle.js]: should this be `77`?

Comment (fvictorio) [reply blockchain/truffle.js]: Yes, sorry, copy-paste error.

**Review by phahulin — APPROVED** [review]

**Comment by fvictorio** [comment]
Fixed. I also re-checked that the app works on Sokol and everything seems to be OK.

**Review by garatortiz — APPROVED** [review]

**Pull request merged**